source/MaterialXRender/Handlers/Mesh.cpp

```diff
@@ -126,4 +128,70 @@ bool Mesh::generateTangents(MeshStreamPtr positionStream, MeshStreamPtr texcoord
     return true;
 }
 
-}
\ No newline at end of file
+void Mesh::mergePartitions()
+{
+    if (getPartitionCount() <= 1)
+    {
+        return;
+    }
+
+    MeshPartitionPtr merged = MeshPartition::create();
+    merged->setIdentifier("merged");
+    for (size_t p = 0; p < getPartitionCount(); p++)
+    {
+        MeshPartitionPtr part = getPartition(p);
+        merged->getIndices().insert(merged->getIndices().end(),
+                                    part->getIndices().begin(),
+                                    part->getIndices().end());
+        merged->setFaceCount(merged->getFaceCount() + part->getFaceCount());
+    }
+
+    _partitions.clear();
+    addPartition(merged);
+}
+
+void Mesh::splitByUdims()
+{
+    MeshStreamPtr texcoords = getStream(MeshStream::TEXCOORD_ATTRIBUTE, 0);
+    if (!texcoords)
+    {
+        return;
+    }
+
+    using UdimMap = std::map<uint32_t, MeshPartitionPtr>;
+    UdimMap udimMap;
+    for (size_t p = 0; p < getPartitionCount(); p++)
+    {
+        MeshPartitionPtr part = getPartition(p);
+        for (size_t f = 0; f < part->getFaceCount(); f++)
+        {
+            uint32_t i0 = part->getIndices()[f * 3 + 0];
+            uint32_t i1 = part->getIndices()[f * 3 + 1];
+            uint32_t i2 = part->getIndices()[f * 3 + 2];
+
+            const Vector2& uv0 = reinterpret_cast<Vector2*>(&texcoords->getData()[0])[i0];
```

Comment:
I don't quite understand this partition by udim's. UV's for only the first vertex in a face is checked and based on its udim index this face is placed into a partition. What if the other two vertices reach into other udim tiles.. or what if a single large face overlap multiple udim tiles?

But I guess I don't quite understand what this partitioning on udim is used for? Since a single face can stretch over many udim tiles it can't be used for binding different udim tile textures to these partitions, right?

Reply:
Since that code is from the viewer fork, I can add some thoughts here.  Ideally, we would check all vertices of each face and emit a warning to the user when faces span multiple UDIMs.  For now, the logic is overly optimistic and assumes the mesh is well formed for UDIM-based rendering.

Reply:
OK. Thanks for clarifying.